A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -320,6 +323,67 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         }
     }
 
+    tryUpdateAltitudeConstraint(force = false) {
+        if (this.flightPlanManager.getIsDirectTo()) {
+            return this.constraintAlt = 0;
+        }
+        const activeWptIdx = this.flightPlanManager.getActiveWaypointIndex();
+        const altLock = Simplane.getAutoPilotSelectedAltitudeLockValue("feet");
+        if (force || activeWptIdx !== this.activeWptIdx || altLock !== this.altLock) {
+            this.activeWptIdx = activeWptIdx;
+            this.altLock = altLock;
+            return this.constraintAlt = this.getAltitudeConstraint();
+        }
+        return this.constraintAlt;
+    }
+
+    getAltitudeConstraint() {
+        const fph = Simplane.getCurrentFlightPhase();
+        const type = fph < FlightPhase.FLIGHT_PHASE_CRUISE || fph === FlightPhase.FLIGHT_PHASE_GOAROUND ? 3 : 2;
+        const rte = this.flightPlanManager.getWaypoints(0);
+        let tmp = 0;
+
+        for (let i = this.activeWptIdx; i < rte.length; i++) {
+            const wpt = rte[i];
+            if (!isFinite(wpt.legAltitude1)) {
+                continue;
+            }
+            // Ensure constraint waypoint after TOD is not a constraint for climb phase
+            if (tmp) {
+                if (isFinite(wpt.legAltitude2)) {
+                    if (type === 3 && (wpt.legAltitude1 > wpt.legAltitude2 || wpt.legAltitude1 > tmp)) {
+                        return 0;
+                    }
+                } else if (type === 3 && wpt.legAltitude1 > tmp) {
+                    return 0;
+                }
+            } else {
+                tmp = wpt.legAltitude1;
+            }
+            if (wpt.legAltitudeDescription === 0) {
+                continue;
+            }
+            if (wpt.legAltitudeDescription === 4) {
+                if (type === 3) {
+                    if (this.altLock < wpt.legAltitude2) {
+                        return wpt.legAltitude2;
+                    }
```

Comment:
Here as well

Reply:
no, would break the logic for the else if. Type 3 must go alone, otherwise I can run into the else if with type 3 as well. Which is no good. Same for the other one.